flutter/lib/desktop/pages/connection_page.dart

```diff
@@ -501,18 +503,67 @@ class _ConnectionPageState extends State<ConnectionPage>
               child: Row(
                 mainAxisAlignment: MainAxisAlignment.end,
                 children: [
-                  Button(
-                    isOutline: true,
-                    onTap: () => onConnect(isFileTransfer: true),
-                    text: "Transfer file",
+                  ElevatedButton(
+                    onPressed: () {
+                      if (selectedConnectionType == 'Transfer file') {
+                        onConnect(isFileTransfer: true);
+                      } else if (selectedConnectionType == 'View camera') {
+                        onConnect(isViewCamera: true);
+                      } else {
+                        onConnect();
+                      }
+                    },
+                    child: Text(selectedConnectionType), // Show current selection
                   ),
-                  const SizedBox(
-                    width: 17,
+                  MenuAnchor(
+                    builder: (context, controller, builder) {
+                      return IconButton(
+                        visualDensity: VisualDensity.compact,
+                        icon: controller.isOpen ? const Icon(Icons.arrow_drop_up) : const Icon(Icons.arrow_drop_down),
+                        onPressed: () {
+                          setState(() {
+                            if (controller.isOpen) {
+                              controller.close();
+                            } else {
+                              controller.open();
+                            }
+                          });
+                        },
+                      );
+                    },
+                    menuChildren: <Widget>[
+                      MenuItemButton(
+                        onPressed: () {
+                          setState(() {
+                            selectedConnectionType = 'Connect';
+                          });
+                          Navigator.pop(context);
+                        },
+                        child: Text('Connect'),
+                      ),
+                      MenuItemButton(
+                        onPressed: () {
+                          setState(() {
+                            selectedConnectionType = 'Transfer file';
+                          });
+                          Navigator.pop(context);
+                        },
+                        child: Text('Transfer file'),
+                      ),
+                      MenuItemButton(
+                        onPressed: () {
+                          setState(() {
+                            selectedConnectionType = 'View camera';
+                          });
+                          Navigator.pop(context);
```

Comment:
Remove `Navigator.pop(context)`, which causing transparent window, and connect directly on click rather than change `selectedConnectionType`.